src/url-params.ts

```diff
@@ -98,6 +101,15 @@ export class AnalyticsUrlParams {
 		return Array.from(res);
 	}
 
+	/**
+	 * @return an array of passed Device Ids via the URL param if there was any
+	 */
+	getPassedDeviceIds() {
+		const passedIds = this.passedDeviceIds;
+		const res = new Set(passedIds);
+		return Array.from(res);
```

Comment:
Maybe it would be better to run the logic once in `setDeviceIds` instead of on every retrieval?
Furthermore, I'm confused about the repeated casting between `Set` and `Array` types - is there any meaning to it?

Reply:
Agreed with Pranas. On a different note, I suggest refactoring this function to return a single device ID, instead of an array. 

We already have `allDeviceIds()` that returns all the device identifiers. Is there a need to return specifically the passed device identifiers?

Reply:
> We already have `allDeviceIds()` that returns all the device identifiers. Is there a need to return specifically the passed device identifiers?

This was exactly my confusion in the comment above.

Reply:
> We already have allDeviceIds() that returns all the device identifiers. Is there a need to return specifically the passed device identifiers?

My reasoning was that for some visits,  the `allDeviceIds()` function gets the total of devices already stored in the client plus those passed via the URL, but there is no way of distinguishing between those once they are all in the same array. So this way, in case there is a need later on, we can differentiate between those already stored by the client and those that were obtain by the d_id param.

Reply:
No need to support an unknown use case now @laziob. We could add that when the need arises. Besides, we are now designing the device linking logic to always prefer the 1st passed device ID over everything else. So it's perfectly fine to offer an opinionated method here (which always returns the 1st passed value).

Reply:
@gelbal @pranasziaukas 

I understand. I just commited the new changes. Please let me know how it looks now.